WIP: Rework identifiers

MarekSuchanek opened the pull request.

ValentinFutterer commented on README.md:
```diff
@@ -209,6 +234,21 @@ For specifying the organisation(s) to which a contact or contributor belongs.
+    <li id="related_identifier_scheme_uri_tree"><a href="#related_identifier_scheme_uri">scheme_uri</a></li>
+    <li id="related_identifier_scheme_type_tree"><a href="#related_identifier_scheme_type">scheme_type</a></li>
+    <li id="related_identifier_resource_type_tree"><a href="#related_identifier_resource_type">resource_type</a></li>
+  </ul>
```
I dont quite get what these are doing here. They are on the same level as DMP and affiliation. Although I also dont know why affiliation is on the same level as DMP and not a field of DMP 😅 Maybe you can explain:
<img width="299" height="380" alt="image" src="https://github.com/user-attachments/assets/dbb9ec2e-206b-4caa-852b-83ebd0feacac" />

The problem I see is, that both alternate and related idnetifiers are also a subfield of DMP - in my opinion there is a clash here, but maybe there is a difference I am not seeing?
<img width="296" height="121" alt="image" src="https://github.com/user-attachments/assets/0f363d9c-f7da-460f-bc48-34ea3e61b071" />

MarekSuchanek replied:
this is a bit tricky... those are shared entities, it can be put to the tree with just simple root (dmp) but then it will repeat multiple times and also the way IDs are done links will need to be reworked

MarekSuchanek commented:
> These comments are a repost from the last review, maybe they were forgotton or not relevant:
> 
> In contributor_id and other fields, the link part from the ORCID example was removed, so just the id remained - maybe this should be done here too to remain consistent.
> 
> ...

I fixed it for ID examples and also fixed Creator ID type to suggested values. I would not change the others (as those are not ID fields and we did not agree in issue on that)... so only ID type should have "suggested values". With that I hope it should be all ready for merging...
